services/endpointsrv/service.go

```diff
@@ -438,17 +438,71 @@ func (e *EndpointService) EndpointCheckStatus(ctx context.Context, checkID strin
 
 	out := make([]types.EndpointCheckStep, len(steps))
 	for idx, step := range steps {
+		status, conclusion := StepStatusAndConclusion(step)
 		out[idx] = types.EndpointCheckStep{
-			StepID:    step.ID,
-			EvalID:    step.EvalID,
-			Input:     []byte(step.Input.String),
-			Output:    []byte(step.Output.String),
-			Assertion: step.Assertion.String,
+			StepID:     step.ID,
+			EvalID:     step.EvalID,
+			Input:      []byte(step.Input.String),
+			Output:     []byte(step.Output.String),
+			Assertion:  step.Assertion.String,
+			Status:     status,
+			Conclusion: conclusion,
 		}
 	}
 
+	status, conclusion := CheckStatusAndConclusion(out)
 	return types.EndpointCheck{
-		CheckID: checkID,
-		Steps:   out,
+		CheckID:    checkID,
+		Steps:      out,
+		Status:     status,
+		Conclusion: conclusion,
 	}, nil
 }
+
+// StepStatusAndConclusion infers the status of a step based on the contents of the database.
+func StepStatusAndConclusion(step db.UnweaveEndpointCheckStep) (types.CheckStatus, *types.CheckConclusion) {
+	if !step.Output.Valid {
+		return types.CheckPending, nil
+	}
+
+	if !step.Assertion.Valid {
+		return types.CheckInProgress, nil
+	}
```

Comment:
**nit** without knowing that this is a database type, and that these are `sql.NullString` it looks like `Valid` is a property of the Output / Assertion (rather than the underlying datatype). 

Maybe we can say: 
```go
hasOutput := step.Output.Valid
hasAssertion := step.Assertion.Valid

if hasOutput {
 //...
}

if hasAssertion {
 //...
}
```

I think this makes it a little clearer that we are testing for _presence_ and not some business domain "this is valid or not" rules.